spinoso-string/src/enc/mod.rs

```diff
@@ -0,0 +1,462 @@
+mod ascii;
+mod binary;
+mod impls;
+mod utf8;
+
+use alloc::vec::Vec;
+use core::ops::Range;
+use core::slice::SliceIndex;
+
+use ascii::AsciiString;
+use binary::BinaryString;
+use utf8::Utf8String;
+
+use crate::codepoints::InvalidCodepointError;
+use crate::encoding::Encoding;
+use crate::iter::{Bytes, IntoIter, Iter, IterMut};
+use crate::ord::OrdError;
+
+#[derive(Clone)]
+pub enum EncodedString {
+    Ascii(AsciiString),
+    Binary(BinaryString),
+    Utf8(Utf8String),
+}
+
+impl Default for EncodedString {
+    fn default() -> Self {
+        Self::Utf8(Utf8String::new(Vec::new()))
+    }
+}
+
+// Constructors
+impl EncodedString {
+    #[inline]
+    #[must_use]
+    pub fn new(buf: Vec<u8>, encoding: Encoding) -> Self {
+        match encoding {
+            Encoding::Ascii => Self::Ascii(AsciiString::new(buf)),
+            Encoding::Binary => Self::Binary(BinaryString::new(buf)),
+            Encoding::Utf8 => Self::Utf8(Utf8String::new(buf)),
+        }
+    }
+}
+
+impl EncodedString {
+    #[inline]
+    #[must_use]
+    pub fn encoding(&self) -> Encoding {
+        match self {
+            EncodedString::Ascii(_) => Encoding::Ascii,
+            EncodedString::Binary(_) => Encoding::Binary,
+            EncodedString::Utf8(_) => Encoding::Utf8,
+        }
+    }
+}
+
+// Defer to Encoded Implementation
+impl EncodedString {
+    #[inline]
+    #[must_use]
+    pub fn into_vec(self) -> Vec<u8> {
+        match self {
+            EncodedString::Ascii(n) => n.into_vec(),
```

Comment:
Sorry to be nit-picky, but can we replace all uses of `n` for the inner bit of the enum with `inner`?

Reply:
This is a very good thing to be nit-picky about. As I mentioned on discord, I have 0 ideas why I used `n` instead of something like `inner` or a shortened version like `i` or `e` etc haha. Easy update!